pkg/controller/studyjob/manifest_parser.go

```diff
@@ -28,30 +27,50 @@ import (
 	"k8s.io/apimachinery/pkg/util/uuid"
 )
 
-func getWorkerManifest(c katibapi.ManagerClient, studyID string, trial *katibapi.Trial, workerSpec *katibv1alpha1.WorkerSpec, kind string, ns string, dryrun bool) (string, *bytes.Buffer, error) {
-	var wtp *template.Template = nil
-	var err error
-	if workerSpec != nil && workerSpec.GoTemplate.RawTemplate != "" {
-		wtp, err = template.New("Worker").Parse(workerSpec.GoTemplate.RawTemplate)
+func getTemplateStrFromConfigMap(cNamespace, cName, tPath string) (string, error) {
+	sjc, err := studyjobclient.NewStudyjobClient(nil)
+	if err != nil {
+		return "", err
+	}
+	return sjc.GetTemplate(cNamespace, cName, tPath)
+}
+
+func getTemplateStr(goTemplate *katibv1alpha1.GoTemplate, getDefaultTemplateSpec func()(string,string,string)) (string, error) {
+	if goTemplate.RawTemplate != "" {
+		return goTemplate.RawTemplate, nil
 	} else {
-		wPath := "defaultWorkerTemplate.yaml"
-		if workerSpec != nil && workerSpec.GoTemplate.TemplatePath != "" {
-			wPath = workerSpec.GoTemplate.TemplatePath
-		}
-		sjc, err := studyjobclient.NewStudyjobClient(nil)
-		if err != nil {
-			return "", nil, err
-		}
-		wtl, err := sjc.GetWorkerTemplates()
-		if err != nil {
-			return "", nil, err
-		}
-		if wt, ok := wtl[wPath]; !ok {
-			return "", nil, fmt.Errorf("No worker template name %s", wPath)
-		} else {
-			wtp, err = template.New("Worker").Parse(wt)
+		tName, tNamespace, tPath := getDefaultTemplateSpec()
+		if goTemplate.TemplateSpec != nil {
+			tName = goTemplate.TemplateSpec.ConfigMapName
```

Comment:
When the field of TemplateSpec is emply, we should use the default value.
For example, we can omit the namespace of ConfigMap when we want to use the same namespace as the studyjobcontroller.